dlhub_sdk/utils/publish.py

```diff
@@ -0,0 +1,244 @@
+import logging
+import json
+import os
+import requests
+import globus_sdk
+
+from funcx.sdk.client import FuncXClient
+from funcx import ContainerSpec
+from time import sleep
+from github import Github
+import base64
+
+import mdf_toolbox
+import urllib
+
+#Uncomment if we are trying to mint identifiers
+# from identifiers_client.identifiers_api import identifiers_client, IdentifierClient
+# from identifiers_client.config import config
+
+logger = logging.getLogger(__name__)
+
+
+def create_container_spec(metadata):
+    """
+    Create the container spec for the Container Service. Iterate through
+    the dependencies and add them. Also include parsl, toolbox, home_run, etc.
+
+    :param metadata:
+    :return:
+    """
+
+    # Get the list of requirements from the schema
+    dependencies = []
+    try:
+        for k, v in metadata['dlhub']['dependencies']['python'].items():
+            dependencies.append("{0}=={1}".format(k, v))
+    except:
+        # There are no python dependencies
+        pass
+
+    model_location = None
+    if 'transfer_method' in metadata['dlhub']:
+        if 'S3' in metadata['dlhub']['transfer_method']:
+            model_location = metadata['dlhub']['transfer_method']['S3']
+        elif 'POST' in metadata['dlhub']['transfer_method']:
+            model_location = metadata['dlhub']['transfer_method']['path']
+    if 'repository' in metadata:
+        model_location = metadata['repository']
+    cs = ContainerSpec(
+        name=metadata['dlhub']['shorthand_name'],
+        pip=dependencies,
+        python_version="3.7",
+        payload_url=model_location,
+        conda=[],
+    )
+
+    return cs
+
+
+def search_ingest(task, header):
+    """
+    Ingest the servable data into a Globus Search index.
+
+    Args:
+        task (dict): the task description.
+    """
+    logger.debug("Ingesting servable into Search.")
+
+#    idx = "dlhub"
+    # idx = '847c9105-18a0-4ffb-8a71-03dd76dfcc9d'
+    iden = "https://dlhub.org/servables/{}".format(task['dlhub']['id'])
+    # index = mdf_toolbox.translate_index(idx)
+
+    ingestable = task
+    d = [convert_dict(ingestable, str)]
+
+    glist = []
+    visible_to = task['dlhub'].get('visible_to', ['public'])
+
+    # Add public so it isn't an empty list
+    if len(visible_to) == 0:
+        visible_to = ['public']
+
+    for document in d:
```

Comment:
a few more comments about how/why you're formatting things would useful (I know you're getting things into GMeta shape for search, but the reader might not). Perhaps linking to the Search docs in the docstring